datajoint/connection.py

```diff
@@ -172,7 +172,7 @@ def __init__(self, host, user, password, port=None, init_fun=None, use_tls=None,
         if use_tls is not False:
             self.conn_info['ssl'] = use_tls if isinstance(use_tls, dict) else {'ssl': {}}
         self.conn_info['ssl_input'] = use_tls
-        self.conn_info['host_input'] = host_input
+        self.conn_info['host_input'] = host_input if host_input else ''
```

Comment:
Yes, this is very much a hack that won't work for connection plugin users if they call `Connection` explicitly. Looks like `get_host_hook` was simply misplaced in `conn` rather than `Connection` in the original implementation. Should be pretty straightforward to patch. Could you:
- remove `dj.Connection` argument `host_input`
- move `get_host_hook` be called first in `dj.Connection`

Reply:
a) not sure how get_host_hook would be called in proposed refactor - simply with empty string?

b) assuming 'a' as yes, the `connect_host_hook` call later depends on conn_info containing 'host_input' item, and it also tries to 'iterate' on this string (if '://' in ...), so unless I'm misunderstanding things, this would just move the problem elsewhere

feel free to counter-PR if that's easier, mainly breaking my 'you break it, you buy it' rule by putting 'something' in to get the ball rolling for just the regression aspect.

Reply:
a. Basically, in the first line of the `dj.Connection` constructor, you would have:
   ```python
   host_input, host = (host, get_host_hook(host_input))
   ```
b. This would not be applicable as `host_input` would always be satisfied now provided `host` is a string (as assumption that has already been there for `dj.Connection`).

Could you also add a test for this too?

Reply:
a. Basically, in the first line of the `dj.Connection` constructor, you would have:
   ```python
   host_input, host = (host, get_host_hook(host_input))
   ```
b. This would not be applicable as `host_input` would always be satisfied now provided `host` is a string (as assumption that has already been there for `dj.Connection`).

Could you also add a test for this too?

Reply:
Let me know if this makes sense or if you prefer I open another one.

Reply:
a) the proposed call to get_host_hook(host_input) will now have no input variable - where does the value for 'host_input' come from if in the proposed call if it is removed from __init__ signature? or do you mean `host_input, host = (host, host)` ? 
b) still confused due to 'a')
c) r.e. tests: this is a hack fix to a regression that I didn't introduce, so don't think this is in-scope here. You break, you buy (tm).

Reply:
Oh sorry, that was a typo. Fixed it above.
OK, I'll open a PR then.